Add a backup guide for the Scalar DB server (#254)
scalar-labs/scalardb: docs/backup-restore.md

That means that you need to create a consistent snapshot by dumping all tables in a single transaction.

> For example, you can use `mysqldump` command with `--single-transaction` option in MySQL and `pg_dump` command in PostgreSQL to achieve that.

Reviewer: BTW, Is this true even when we use ConsensusCommit?

Author: Thank you for reviewing this!

Yes. `mysqldump` command with `--single-transaction` and `pg_dump` command can take a backup that's automatically recoverable to a transactionally-consistent state. So, it's true even when we use ConsensusCommit.

Or when you use Amazon RDS (Relational Database Service) or Azure Database for MySQL/PostgreSQL, you can restore to any point within the backup retention period with the automated backup feature, which satisfies the requirement.

## Scalar DB server

> Since the Scalar DB server implements [scalar-admin](https://github.com/scalar-labs/scalar-admin) interface, we can easily pause/unpause it as mentioned in [Cassandra](#Cassandra). 

Reviewer: ```suggestion
Since Scalar DB server implements [scalar-admin](https://github.com/scalar-labs/scalar-admin) interface, you can easily pause the server to stop accepting incoming requests and drain in-flight requests.
```

> So we can take a transactionally-consistent backup for Scalar DB by pausing the Scalar DB server, taking backups of all the tables, and unpausing the Scalar DB server.

Reviewer: ```suggestion
You can use paused duration and choose one of the following ways to make your databases transactionally-consistent even after being restored depending on the underlining database.
* Take backups of your databases during the paused duration with snapshot feature.
* Restore to a point in the paused duration with point-in-time-restore (PITR) feature.
```

> You can use paused duration and choose one of the following ways to make your databases transactionally-consistent even after being restored depending on the underlining database.

Reviewer: ```suggestion
You can use paused duration and choose one of the following ways depending on the underlining database to make your databases transactionally-consistent even after restored.
```
Sorry, I re-looked at it and found flaws so fixed it.

> If you use Cassandra, you can use [Cassy](https://github.com/scalar-labs/cassy) integrated with `scalar-admin`.

Reviewer: ```suggestion
```

> Please see [the doc](https://github.com/scalar-labs/cassy/blob/master/docs/getting-started.md#take-cluster-wide-consistent-backups) for more details.

Reviewer: ```suggestion
```
